### PR TITLE
test: enable Elastic Backend IT tests

### PR DESCRIPTION
**Pull request by kuisathaverat**
This PR enables the tests related to the logs for the backend Elastic. in the process, we have found two bugs, one related to the index used, and the other related to the label ci_pipeline_id

closes to https://github.com/jenkinsci/opentelemetry-plugin/issues/783 https://github.com/jenkinsci/opentelemetry-plugin/issues/683 https://github.com/jenkinsci/opentelemetry-plugin/issues/665

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/opentelemetry/backend/GrafanaBackend.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
 import java.util.Objects;
 import java.util.Optional;
 
-public class GrafanaBackend extends ObservabilityBackend implements TemplateBindingsProvider {
+public class GrafanaBackend extends ObservabilityBackend {
```
Comment (kuisathaverat) [inline src/main/java/io/jenkins/plugins/opentelemetry/backend/GrafanaBackend.java:32]: redundant

File: src/main/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticLogsBackend.java [hunk 2]
```diff
@@ -5,31 +5,31 @@
...
-import java.io.UnsupportedEncodingException;
-import java.net.URLEncoder;
-import java.nio.charset.StandardCharsets;
-import java.util.Collections;
-import java.util.Map;
-import java.util.logging.Level;
-import java.util.logging.Logger;
 
-public abstract class ElasticLogsBackend extends AbstractDescribableImpl<ElasticLogsBackend> implements Describable<ElasticLogsBackend>, ExtensionPoint {
+public abstract class ElasticLogsBackend extends AbstractDescribableImpl<ElasticLogsBackend> implements ExtensionPoint {
```
Comment (kuisathaverat) [inline src/main/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticLogsBackend.java:32]: redundant

File: src/main/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticsearchFields.java [hunk 3]
```diff
@@ -21,7 +21,7 @@ public interface ElasticsearchFields {
     String FIELD_CI_PIPELINE_ID = "labels." + JenkinsOtelSemanticAttributes.CI_PIPELINE_ID.getKey().replace('.', '_');
     String FIELD_CI_PIPELINE_RUN_NUMBER = "numeric_labels." + JenkinsOtelSemanticAttributes.CI_PIPELINE_RUN_NUMBER.getKey().replace('.', '_');
     String FIELD_JENKINS_STEP_ID = "labels." + JenkinsOtelSemanticAttributes.JENKINS_STEP_ID.getKey().replace('.', '_');
-    String INDEX_TEMPLATE_PATTERNS = "logs-apm.app-*";
+    String INDEX_TEMPLATE_PATTERNS = "logs-apm.app-*,.ds-logs-apm.app*";
```
Comment (kuisathaverat) [inline src/main/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticsearchFields.java:24]: new index pattern

File: src/main/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticsearchLogStorageRetriever.java [hunk 4]
```diff
@@ -231,7 +231,7 @@ public LogsQueryResult stepLog(@NonNull String jobFullName, int runNumber, @NonN
      * Example of a successful check:
      * <pre>{@code
      * OK: Verify existence of the Elasticsearch Index Template 'logs-apm.app' used to store Jenkins pipeline logs...
-     * OK: Connected to Elasticsearch https://***.europe-west1.gcp.cloud.es.io:9243 with user 'jenkins'.
+     * OK: Connected to Elasticsearch https://***.es.example.com with user 'jenkins'.
```
Comment (kuisathaverat) [inline src/main/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticsearchLogStorageRetriever.java:234]: follow docs best practices, see https://datatracker.ietf.org/doc/html/rfc2606

File: src/main/java/io/jenkins/plugins/opentelemetry/job/log/FlowNodeTraceContext.java [hunk 5]
```diff
@@ -25,7 +25,7 @@ public static FlowNodeTraceContext newFlowNodeTraceContext(@NonNull Run run, @No
         String spanId = span.getSpanContext().getSpanId();
         String traceId = span.getSpanContext().getTraceId();
         Map<String, String> w3cTraceContext = OtelUtils.getW3cTraceContext(span);
-        return new FlowNodeTraceContext(run.getFullDisplayName(), run.getNumber(), flowNode.getId(), traceId, spanId, w3cTraceContext);
+        return new FlowNodeTraceContext(run.getParent().getFullName(), run.getNumber(), flowNode.getId(), traceId, spanId, w3cTraceContext);
```
Comment (kuisathaverat) [inline src/main/java/io/jenkins/plugins/opentelemetry/job/log/FlowNodeTraceContext.java:28]: see https://github.com/jenkinsci/opentelemetry-plugin/issues/783

File: src/main/java/io/jenkins/plugins/opentelemetry/job/log/OtelLogStorage.java [hunk 6]
```diff
@@ -160,15 +160,15 @@ public AnnotatedLargeText<FlowExecutionOwner.Executable> overallLog(@NonNull Flo
         }
 
         Span span = tracer.spanBuilder("OtelLogStorage.overallLog")
-            .setAttribute(JenkinsOtelSemanticAttributes.CI_PIPELINE_ID, run.getFullDisplayName())
+            .setAttribute(JenkinsOtelSemanticAttributes.CI_PIPELINE_ID, run.getParent().getFullName())
```
Comment (kuisathaverat) [inline src/main/java/io/jenkins/plugins/opentelemetry/job/log/OtelLogStorage.java:163]: see https://github.com/jenkinsci/opentelemetry-plugin/issues/783

File: src/main/java/io/jenkins/plugins/opentelemetry/job/log/OtelLogStorage.java [hunk 7]
```diff
@@ -160,15 +160,15 @@ public AnnotatedLargeText<FlowExecutionOwner.Executable> overallLog(@NonNull Flo
...
             .setAttribute(JenkinsOtelSemanticAttributes.CI_PIPELINE_RUN_NUMBER, (long) run.getNumber())
             .setAttribute("complete", complete)
             .startSpan();
         try (Scope scope = span.makeCurrent()){
             LogStorageRetriever logStorageRetriever = getLogStorageRetriever();
             Instant startTime = Instant.ofEpochMilli(run.getStartTimeInMillis());
             Instant endTime = run.getDuration() == 0 ? null : startTime.plusMillis(run.getDuration());
-            LogsQueryResult logsQueryResult = logStorageRetriever.overallLog(run.getFullDisplayName(), run.getNumber(), runTraceContext.getTraceId(), runTraceContext.getSpanId(), complete, startTime, endTime);
+            LogsQueryResult logsQueryResult = logStorageRetriever.overallLog(run.getParent().getFullName(), run.getNumber(), runTraceContext.getTraceId(), runTraceContext.getSpanId(), complete, startTime, endTime);
```
Comment (kuisathaverat) [inline src/main/java/io/jenkins/plugins/opentelemetry/job/log/OtelLogStorage.java:171]: see https://github.com/jenkinsci/opentelemetry-plugin/issues/783

File: src/main/java/io/jenkins/plugins/opentelemetry/job/log/OtelLogStorage.java [hunk 8]
```diff
@@ -201,7 +201,7 @@ public AnnotatedLargeText<FlowNode> stepLog(@NonNull FlowNode flowNode, boolean
             LogStorageRetriever logStorageRetriever = getLogStorageRetriever();
             Instant startTime = Instant.ofEpochMilli(run.getStartTimeInMillis());
             Instant endTime = run.getDuration() == 0 ? null : startTime.plusMillis(run.getDuration());
-            LogsQueryResult logsQueryResult = logStorageRetriever.stepLog(run.getFullDisplayName(), run.getNumber(), flowNode.getId(), traceId, spanId, complete, startTime, endTime);
+            LogsQueryResult logsQueryResult = logStorageRetriever.stepLog(run.getParent().getFullName(), run.getNumber(), flowNode.getId(), traceId, spanId, complete, startTime, endTime);
```
Comment (kuisathaverat) [inline src/main/java/io/jenkins/plugins/opentelemetry/job/log/OtelLogStorage.java:204]: see https://github.com/jenkinsci/opentelemetry-plugin/issues/783

File: src/test/java/io/jenkins/plugins/opentelemetry/jcasc/ConfigurationAsCodeElasticLogsBackendExclusiveTest.java [hunk 9]
```diff
@@ -42,7 +42,7 @@ public void should_support_configuration_as_code() {
         MatcherAssert.assertThat(configuration.getEndpoint(), CoreMatchers.is("http://otel-collector-contrib:4317"));
 
         ElasticBackend elastic = (ElasticBackend) configuration.getObservabilityBackends().get(0);
-        MatcherAssert.assertThat(elastic.getKibanaBaseUrl(), CoreMatchers.is("https://kibana.europe-west1.gcp.cloud.es.io:9243"));
+        MatcherAssert.assertThat(elastic.getKibanaBaseUrl(), CoreMatchers.is("https://kibana.es.example.com"));
```
Comment (kuisathaverat) [inline src/test/java/io/jenkins/plugins/opentelemetry/jcasc/ConfigurationAsCodeElasticLogsBackendExclusiveTest.java:45]: follow docs best practices, see https://datatracker.ietf.org/doc/html/rfc2606

File: src/test/java/io/jenkins/plugins/opentelemetry/jcasc/ConfigurationAsCodeElasticLogsBackendTest.java [hunk 10]
```diff
@@ -42,12 +42,12 @@ public void should_support_configuration_as_code() {
         MatcherAssert.assertThat(configuration.getEndpoint(), CoreMatchers.is("http://otel-collector-contrib:4317"));
 
         ElasticBackend elastic = (ElasticBackend) configuration.getObservabilityBackends().get(0);
```
Comment (kuisathaverat) [inline src/test/java/io/jenkins/plugins/opentelemetry/jcasc/ConfigurationAsCodeElasticLogsBackendTest.java:44]: follow docs best practices, see https://datatracker.ietf.org/doc/html/rfc2606

File: src/test/java/io/jenkins/plugins/opentelemetry/jcasc/ConfigurationAsCodeElasticLogsBackendTest.java [hunk 11]
```diff
@@ -42,12 +42,12 @@ public void should_support_configuration_as_code() {
...
-        MatcherAssert.assertThat(elastic.getKibanaBaseUrl(), CoreMatchers.is("https://kibana.europe-west1.gcp.cloud.es.io:9243"));
+        MatcherAssert.assertThat(elastic.getKibanaBaseUrl(), CoreMatchers.is("https://kibana.es.example.com"));
         MatcherAssert.assertThat(elastic.getName(), CoreMatchers.is("My Elastic"));
 
         ElasticLogsBackendWithJenkinsVisualization elasticLogsBackend = (ElasticLogsBackendWithJenkinsVisualization) elastic.getElasticLogsBackend();
         MatcherAssert.assertThat(elasticLogsBackend.getElasticsearchCredentialsId(), CoreMatchers.is("elasticsearch-logs-creds"));
-        MatcherAssert.assertThat(elasticLogsBackend.getElasticsearchUrl(), CoreMatchers.is("https://es.europe-west1.gcp.cloud.es.io:9243"));
+        MatcherAssert.assertThat(elasticLogsBackend.getElasticsearchUrl(), CoreMatchers.is("https://es.es.example.com"));
```
Comment (kuisathaverat) [inline src/test/java/io/jenkins/plugins/opentelemetry/jcasc/ConfigurationAsCodeElasticLogsBackendTest.java:50]: follow docs best practices, see https://datatracker.ietf.org/doc/html/rfc2606

File: src/test/java/io/jenkins/plugins/opentelemetry/jcasc/ConfigurationAsCodeElasticTest.java [hunk 12]
```diff
@@ -33,11 +33,11 @@ public class ConfigurationAsCodeElasticTest {
     public void should_support_configuration_as_code() {
         final JenkinsOpenTelemetryPluginConfiguration configuration = GlobalConfiguration.all().get(JenkinsOpenTelemetryPluginConfiguration.class);
 
-        MatcherAssert.assertThat(configuration.getEndpoint(), CoreMatchers.is("https://my-deployment.apm.europe-west1.gcp.cloud.es.io"));
+        MatcherAssert.assertThat(configuration.getEndpoint(), CoreMatchers.is("https://my-deployment.otel.example.com"));
```
Comment (kuisathaverat) [inline src/test/java/io/jenkins/plugins/opentelemetry/jcasc/ConfigurationAsCodeElasticTest.java:36]: follow docs best practices, see https://datatracker.ietf.org/doc/html/rfc2606

File: src/test/java/io/jenkins/plugins/opentelemetry/jcasc/ConfigurationAsCodeElasticTest.java [hunk 13]
```diff
@@ -33,11 +33,11 @@ public class ConfigurationAsCodeElasticTest {
...
         MatcherAssert.assertThat(configuration.getObservabilityBackends().size(), CoreMatchers.is(1));
 
         ElasticBackend elastic = (ElasticBackend) configuration.getObservabilityBackends().get(0);
-        MatcherAssert.assertThat(elastic.getKibanaBaseUrl(), CoreMatchers.is("https://my-deployment.europe-west1.gcp.cloud.es.io:9243"));
+        MatcherAssert.assertThat(elastic.getKibanaBaseUrl(), CoreMatchers.is("https://my-deployment.es.example.com"));
```
Comment (kuisathaverat) [inline src/test/java/io/jenkins/plugins/opentelemetry/jcasc/ConfigurationAsCodeElasticTest.java:40]: follow docs best practices, see https://datatracker.ietf.org/doc/html/rfc2606

File: src/test/resources/io/jenkins/plugins/opentelemetry/jcasc/elastic-expected.yml [hunk 14]
```diff
@@ -3,14 +3,14 @@ authentication:
     tokenId: "apm-server-token"
 configurationProperties: "otel.exporter.otlp.protocol=grpc"
 disabledResourceProviders: "io.opentelemetry.instrumentation.resources.ProcessResourceProvider"
-endpoint: "https://my-deployment.apm.europe-west1.gcp.cloud.es.io"
+endpoint: "https://my-deployment.otel.example.com"
```
Comment (kuisathaverat) [inline src/test/resources/io/jenkins/plugins/opentelemetry/jcasc/elastic-expected.yml:6]: follow docs best practices, see https://datatracker.ietf.org/doc/html/rfc2606

File: src/test/resources/io/jenkins/plugins/opentelemetry/jcasc/elastic-expected.yml [hunk 15]
```diff
@@ -3,14 +3,14 @@ authentication:
...
 exportOtelConfigurationAsEnvironmentVariables: false
 exporterIntervalMillis: 60000
 exporterTimeoutMillis: 30000
 ignoredSteps: "dir,echo,isUnix,pwd,properties"
 observabilityBackends:
 - elastic:
-    kibanaBaseUrl: "https://my-deployment.europe-west1.gcp.cloud.es.io:9243"
+    kibanaBaseUrl: "https://my-deployment.es.example.com"
```
Comment (kuisathaverat) [inline src/test/resources/io/jenkins/plugins/opentelemetry/jcasc/elastic-expected.yml:13]: follow docs best practices, see https://datatracker.ietf.org/doc/html/rfc2606

File: src/test/resources/io/jenkins/plugins/opentelemetry/jcasc/elastic-logs-exclusive.yml [hunk 16]
```diff
@@ -4,5 +4,5 @@ unclassified:
     observabilityBackends:
       - elastic:
           name: "My Elastic"
-          kibanaBaseUrl: "https://kibana.europe-west1.gcp.cloud.es.io:9243"
+          kibanaBaseUrl: "https://kibana.es.example.com"
```
Comment (kuisathaverat) [inline src/test/resources/io/jenkins/plugins/opentelemetry/jcasc/elastic-logs-exclusive.yml:7]: follow docs best practices, see https://datatracker.ietf.org/doc/html/rfc2606

File: src/test/resources/io/jenkins/plugins/opentelemetry/jcasc/elastic-logs-expected.yml [hunk 17]
```diff
@@ -8,8 +8,8 @@ observabilityBackends:
     elasticLogsBackend:
       elasticLogsBackendWithJenkinsVisualization:
         elasticsearchCredentialsId: "elasticsearch-logs-creds"
-        elasticsearchUrl: "https://es.europe-west1.gcp.cloud.es.io:9243"
-    kibanaBaseUrl: "https://kibana.europe-west1.gcp.cloud.es.io:9243"
+        elasticsearchUrl: "https://es.es.example.com"
```
Comment (kuisathaverat) [inline src/test/resources/io/jenkins/plugins/opentelemetry/jcasc/elastic-logs-expected.yml:11]: follow docs best practices, see https://datatracker.ietf.org/doc/html/rfc2606

File: src/test/resources/io/jenkins/plugins/opentelemetry/jcasc/elastic-logs.yml [hunk 18]
```diff
@@ -14,8 +14,8 @@ unclassified:
     observabilityBackends:
       - elastic:
           name: "My Elastic"
-          kibanaBaseUrl: "https://kibana.europe-west1.gcp.cloud.es.io:9243"
+          kibanaBaseUrl: "https://kibana.es.example.com"
```
Comment (kuisathaverat) [inline src/test/resources/io/jenkins/plugins/opentelemetry/jcasc/elastic-logs.yml:17]: follow docs best practices, see https://datatracker.ietf.org/doc/html/rfc2606

File: src/test/resources/io/jenkins/plugins/opentelemetry/jcasc/elastic.yml [hunk 19]
```diff
@@ -13,14 +13,14 @@ unclassified:
     authentication:
       bearerTokenAuthentication:
         tokenId: "apm-server-token"
-    endpoint: "https://my-deployment.apm.europe-west1.gcp.cloud.es.io"
+    endpoint: "https://my-deployment.otel.example.com"
```
Comment (kuisathaverat) [inline src/test/resources/io/jenkins/plugins/opentelemetry/jcasc/elastic.yml:16]: follow docs best practices, see https://datatracker.ietf.org/doc/html/rfc2606

File: src/test/resources/io/jenkins/plugins/opentelemetry/jcasc/elastic.yml [hunk 20]
```diff
@@ -13,14 +13,14 @@ unclassified:
...
     configurationProperties: "otel.exporter.otlp.protocol=grpc"
     exportOtelConfigurationAsEnvironmentVariables: false
     exporterIntervalMillis: 60000
     exporterTimeoutMillis: 30000
     observabilityBackends:
       - elastic:
           name: "My Elastic"
-          kibanaBaseUrl: "https://my-deployment.europe-west1.gcp.cloud.es.io:9243"
+          kibanaBaseUrl: "https://my-deployment.es.example.com"
```
Comment (kuisathaverat) [inline src/test/resources/io/jenkins/plugins/opentelemetry/jcasc/elastic.yml:24]: follow docs best practices, see https://datatracker.ietf.org/doc/html/rfc2606

File: pom.xml [hunk 21]
```diff
@@ -453,6 +453,15 @@
                     </loggers>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <systemPropertyVariables>
+                        <jenkins.test.timeout>600</jenkins.test.timeout>
```
Comment (kuisathaverat) [inline pom.xml:461]: force timeout for tests to 10 min, JUnit annotations and @Classrule do not work in Jenkins tests there is a monitor in the JenkinsRule class that only changes its timeout with this property.

File: src/main/java/io/jenkins/plugins/opentelemetry/backend/ElasticBackend.java [hunk 22]
```diff
@@ -281,8 +282,8 @@ public FormValidation doCheckKibanaBaseUrl(@QueryParameter("kibanaBaseUrl") Stri
                 return FormValidation.ok();
             }
             try {
-                new URL(kibanaBaseUrl);
-            } catch (MalformedURLException e) {
+                new URI(kibanaBaseUrl).toURL();
```
Comment (kuisathaverat) [inline src/main/java/io/jenkins/plugins/opentelemetry/backend/ElasticBackend.java:285]: `new URL(url)` is deprecated

File: src/test/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticStackIT.java [hunk 23]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        GlobalOpenTelemetry.resetForTest();
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        if (!isInitialized){
```
Comment (kuisathaverat) [inline src/test/java/io/jenkins/plugins/opentelemetry/backend/elastic/ElasticStackIT.java:47]: workaround for JUnit 4 to initialize the Elastic Stack only once in the Class contexts. In JUnit 5 you can use `@BeforeAll` instead.

File: src/test/resources/kibana/kibana.yml [hunk 24]
```diff
@@ -4,6 +4,10 @@ server.host: "0.0.0.0"
 
 telemetry.enabled: false
 
+logging:
```
Comment (kuisathaverat) [inline src/test/resources/kibana/kibana.yml:7]: reduce log lebel in Kibana

File: src/test/resources/elastic-stack.yml [hunk 25]
```diff
@@ -74,9 +74,9 @@ services:
...
-            # KIBANA_USERNAME: "${KIBANA_ES_USER:-admin}"
-            # KIBANA_PASSWORD: "${KIBANA_ES_PASS:-changeme}"
+            KIBANA_USERNAME: "${KIBANA_ES_USER:-admin}"
+            KIBANA_PASSWORD: "${KIBANA_ES_PASS:-changeme}"
             FLEET_SERVER_INSECURE_HTTP: "1"
             FLEET_SERVER_ELASTICSEARCH_INSECURE: "1"
-            FLEET_SERVER_SERVICE_TOKEN: AAEAAWVsYXN0aWMvZmxlZXQtc2VydmVyL2VsYXN0aWMtcGFja2FnZS1mbGVldC1zZXJ2ZXItdG9rZW46bmgtcFhoQzRRQ2FXbms2U0JySGlWQQ
-            KIBANA_FLEET_SERVICE_TOKEN: AAEAAWVsYXN0aWMvZmxlZXQtc2VydmVyL2VsYXN0aWMtcGFja2FnZS1mbGVldC1zZXJ2ZXItdG9rZW46bmgtcFhoQzRRQ2FXbms2U0JySGlWQQ
+            # FLEET_SERVER_SERVICE_TOKEN: AAEAAWVsYXN0aWMvZmxlZXQtc2VydmVyL2VsYXN0aWMtcGFja2FnZS1mbGVldC1zZXJ2ZXItdG9rZW46bmgtcFhoQzRRQ2FXbms2U0JySGlWQQ
+            # KIBANA_FLEET_SERVICE_TOKEN: AAEAAWVsYXN0aWMvZmxlZXQtc2VydmVyL2VsYXN0aWMtcGFja2FnZS1mbGVldC1zZXJ2ZXItdG9rZW46bmgtcFhoQzRRQ2FXbms2U0JySGlWQQ
```
Comment (kuisathaverat) [inline src/test/resources/elastic-stack.yml:82]: back to use username and password

**Review by v1v — APPROVED** [review]

**Pull request merged**